src/components/profile/user/index.js

```diff
@@ -1,85 +1,132 @@
+import { Paper, CircularProgress } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
 
 // Icons
-import KeyboardReturn from '@material-ui/icons/KeyboardReturn';
+import * as Icon from '@material-ui/icons';
 
-// Redux
-import { connect } from 'react-redux';
+// Context
 import {
-  uploadImageAction,
-  logoutUserAction,
-} from '../../../redux/actions/userActions';
+  useUserLogout,
+  useCurrentUserData,
+  useUploadImageData,
+} from '../userContext';
+
+const useStyles = makeStyles(theme => ({
+  paper: {
+    padding: 20,
+  },
+  profile: {
+    '& .image-wrapper': {
+      textAlign: 'center',
+      position: 'relative',
+      '& button': {
+        position: 'absolute',
+        top: '80%',
+        left: '70%',
+      },
+    },
+    '& .profile-image': {
+      width: 200,
+      height: 'auto',
+      objectFit: 'cover',
+      maxWidth: '100%',
+      borderRadius: '50%',
+    },
+    '& .profile-details': {
+      textAlign: 'center',
+      '& span, svg': {
+        verticalAlign: 'middle',
+      },
+      '& a': {
+        color: theme.palette.primary.main,
+      },
+    },
+    '& hr': {
+      border: 'none',
+      margin: '0 0 10px 0',
+    },
+    '& svg.button': {
+      '&:hover': {
+        cursor: 'pointer',
+      },
+    },
+  },
+  spinnerDiv: {
+    textAlign: 'center',
+    marginTop: 50,
+    marginBottom: 50,
+  },
+}));
 
 /** View component for displaying the user's profile
  * @type {React.FunctionComponent}
- * @param {object} props
- * @param {object} props.classes
- * @param {string} props.user.handle
- * @param {string} props.user.createdAt
- * @param {object} props.user.bio
- * @param {any} props.logoutUserAction
- * @param {any} props.uploadImageAction
  */
-const UserProfileDisplayView = ({
-  classes,
-  user: {
-    userInfo: { handle, createdAt, bio = {} },
-  },
-  logoutUserAction,
-  uploadImageAction,
-}) => {
+export const UserProfileDisplay = () => {
+  const classes = useStyles();
+  const { currentUser } = useCurrentUserData();
+  const { logoutUser } = useUserLogout();
+  const { uploadImage } = useUploadImageData();
+
   const handleEditPhoto = () => {
     const fileInput = document.getElementById('imageUpload');
     fileInput.click();
   };
 
   const handleImageChange = event => {
     const image = event.target.files[0];
-    const formData = new FormData();
-    formData.append('image', image, image.name);
-    uploadImageAction(formData, handle);
+    if (image.size / 1024 > 50000) {
+      toast.error('This image size exceeds the acceptable limit', {
+        position: toast.POSITION.BOTTOM_LEFT,
+        autoClose: 8000,
+      });
+    } else {
+      const formData = new FormData();
+      formData.append('image', image, image.name);
+      uploadImage(formData, currentUser?.handle);
+    }
   };
 
-  const handleLogout = () => logoutUserAction();
-
+  const handleLogout = () => logoutUser();
+  if (Object.keys(currentUser)?.length > 0) {
```

Comment:
It'd be faster if you just checked 
`if(currentUser?.handle||currentUser?.createdAt)`
rather than cycling through all keys on the object

Reply:
Fixed